sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/MetricAnomalyAlertSnoozeCondition.java

```diff
@@ -1,89 +1,90 @@
-    private int autoSnooze;
-
-    /*
-     * snooze scope
-     */
-    @JsonProperty(value = "snoozeScope", required = true)
+    private Integer autoSnooze;
     private SnoozeScope snoozeScope;
+    private Boolean onlyForSuccessive;
 
-    /*
-     * only snooze for successive anomalies
+    /**
+     * Create an instance of MetricAnomalyAlertSnoozeCondition describing anomaly alert snooze configuration.
+     *
+     * @param autoSnooze the snooze data point count.
+     * @param snoozeScope the scope in which anomalies to be snoozed may appear.
+     * @param onlyForSuccessive true to snooze successive anomalies, false otherwise.
      */
-    @JsonProperty(value = "onlyForSuccessive", required = true)
-    private boolean onlyForSuccessive;
+    public MetricAnomalyAlertSnoozeCondition(int autoSnooze,
+                                             SnoozeScope snoozeScope,
+                                             boolean onlyForSuccessive) {
+        this.autoSnooze = autoSnooze;
+        this.snoozeScope = snoozeScope;
+        this.onlyForSuccessive = onlyForSuccessive;
+    }
 
     /**
-     * Get the autoSnooze property: snooze point count, value range : [0, +∞).
+     * Gets the snooze data point count.
      *
-     * @return the autoSnooze value.
+     * @return the snooze data point count.
      */
-    public int getAutoSnooze() {
+    public Integer getAutoSnooze() {
         return this.autoSnooze;
     }
 
     /**
-     * Set the autoSnooze property: snooze point count, value range : [0, +∞).
+     * Gets the scope in which anomalies to be snoozed may appear.
      *
-     * @param autoSnooze the autoSnooze value to set.
-     * @return the MetricAnomalyAlertSnoozeCondition object itself.
+     * @return the scope.
      */
-    public MetricAnomalyAlertSnoozeCondition setAutoSnooze(int autoSnooze) {
-        this.autoSnooze = autoSnooze;
-        return this;
+    public SnoozeScope getSnoozeScope() {
+        return this.snoozeScope;
     }
 
     /**
-     * Get the snoozeScope property: snooze scope.
+     * Gets boolean indicating whether to snooze successive anomalies or not.
      *
-     * @return the snoozeScope value.
+     * @return the onlyForSuccessive value.
      */
-    public SnoozeScope getSnoozeScope() {
-        return this.snoozeScope;
+    public Boolean isOnlyForSuccessive() {
+        return this.onlyForSuccessive;
     }
 
     /**
-     * Set the snoozeScope property: snooze scope.
+     * Sets the value indicating the snooze point count, the value range is : [0, +∞).
      *
-     * @param snoozeScope the snoozeScope value to set.
+     * @param autoSnooze the the snooze point count.
      * @return the MetricAnomalyAlertSnoozeCondition object itself.
      */
-    public MetricAnomalyAlertSnoozeCondition setSnoozeScope(SnoozeScope snoozeScope) {
-        this.snoozeScope = snoozeScope;
+    public MetricAnomalyAlertSnoozeCondition setAutoSnooze(Integer autoSnooze) {
```

Comment:
Should this property also have a setter in spite of being a required param in the constructor?

Reply:
Discussed offline; this is a shared model for create|get|update APIs hence we expose setters to enable update use-cases.